R/read_csv.R

```diff
@@ -159,32 +161,62 @@ read_sample_info_csv <- function(csv_file) {
   csv_file_info
 }
 
-#' Read sampling results from the supplied CSV files
-#'
-#' Read sampling results from the supplied CSV files. Returns a list containing
-#' sampling arguments, the diagonal of the inverse mass matrix, the post-warmup
-#' samples, the sampling parameters and optionally samples produced during
-#' warmup.
+#' Read samples from CmdStan CSV files into \R
 #'
 #' @export
-#' @param output_files Paths to the CSV files to read.
+#' @param files Paths to the CSV files to read.
+#' @param parameters (list) When `NULL`, the function will return
+#' draws for all model parameters. When supplied a list, the function
+#' will return  draws from the listed model parameters.
+#' @param sampler_diagnostics (list) When `NULL`, the function will return
+#' draws for all sampler diagnostics . When supplied a list, the function
+#' will return  draws from the listed sampler diagnostics .
```

Comment:
Is there a reason that `parameters` and `sampler_diagnostics` are lists instead of character vectors (i.e., why `list("x", "y", "z")` and not `c("x", y", "z")`? If there's not a particular reason, then I think character vectors would be more standard in R (and I think most of the code you already have dealing with the lists internally will still work with character vectors).

Reply:
I wanted to support
- NULL which returns all parameters
- list() which returns no parameters
- list("a", "b"...) 

NULL or c() is the same. We could use NULL, NA and c(...) if that would be better.

Reply:
Oh I see. `NA` could be ok. Another option is an empty string, for example, 

```r
sampler_diagnostics = "" # none
sampler_diagnostics = c("divergent__", "treedepth__") # selection
sampler_diagnostics = NULL # all
```

Also, pinging @paul-buerkner since I trust his take on what is sensible from an R perspective. Paul, any opinion on this?

Reply:
I like empty string. No parameter can match that. Will do that.